contents/tutorials/angular-analytics.md

```diff
@@ -0,0 +1,264 @@
+import posthog from 'posthog-js';
+
+//... routes, @NgModule
+
+export class AppRoutingModule {
+  constructor(private router: Router) {
+    this.router.events.subscribe(event => {
+      if (event instanceof NavigationEnd) {
+        posthog.capture('$pageview');
+      }
+    });
+  }
+ }
+```
+
+Now, every time a user moves between pages, PostHog captures a `$pageview` event, not just the first page load. 
+
+Lastly, go back to `main.ts` and make sure to set `capture_pageview` in the PostHog initialization config to `false`. This turns off autocaptured pageviews and ensures you won’t double-capture pageviews on the first load.
+
+```js
+posthog.init(
+  '<ph_project_api_key>',
+  {
+    api_host:'<instance_address>',
+    capture_pageview: false
+  }
+)
+```
+
+## Capturing custom events
+
+Beyond pageviews, there might be more events you want to capture. To do this, you can capture custom events with PostHog. 
+
+To showcase this, add a button to the home component that captures a `home_button_clicked` event in PostHog. To do this, go to your `home.component.html` and add the button with a click handler.
+
+```html
+<h1>Home</h1>
+<button (click)="onClick()">Click me!</button>
+<a routerLink="/about" routerLinkActive="active" ariaCurrentWhenActive="page">Go to About</a>
+```
+
+Afterwards, in `home.component.ts`, import PostHog and add the `onClick()` function with the `posthog.capture` call. Also, you can include a `button_name` property in the capture and use it for filtering later.
+
+```js
+import { Component } from '@angular/core';
+import posthog from 'posthog-js'
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.css']
+})
+export class HomeComponent {
+  onClick() {
+    posthog.capture(
+      'home_button_clicked', 
+      { 'button_name': 'home' }
+    )
+  }
+}
+```
+
+When users click this button, PostHog captures a custom `home_button_clicked` event.
+
+## Setting up and using feature flags
+
+PostHog enables you to use feature flags to control the release of features and code. We can use one to control the text on our button.
+
+To start, go [feature flags tab](https://app.posthog.com/feature_flags) in PostHog and click "New feature flag." Add a key (like `test-flag`) and set the release conditions to 100% of users. Add any other details, and then press "Save."
+
+![Feature flag](../images/tutorials/angular-analytics/flag.png)
+
+To implement our flag, we must change the `home.component.ts` file:
+
+- Add the `buttonText` as a variable in our `HomeComponent` class.
+
+- Import `ChangeDetectorRef` and use a `constructor` to set it up as a service.
+
+- Use the `ngOnInit()` lifecycle hook to check the flags.
+
+- In `ngOnInit()`, wait for flags to load with `posthog.onFeatureFlags` then check our flag with `posthog.isFeatureEnabled('test-flag')`.
+
+- If enabled, change the `buttonText` value and use the `changeDetector` to update the variable.
```

Comment:
Was quite hard to follow this section. maybe its because the order is weird.

Shouldnt it be:

Fetch flag, change button?

Currently its:
Create variable for button
Fetch flag
Change button text.

I think the text could be clearer too

Reply:
Rewrote